File: Python/beginner/1179.py
# https://judge.beecrowd.com/pt/problems/view/1179
# Preenchimento de Vetor IV

# Creating a list to store the entries
entries = []

# Collecting 15 entries
for i in range(15):
    entries.append(int(input()))
    

# Initializing two lists for even and odd numbers
even_numbers = []
odd_numbers = []

# Distributing entries into even and odd lists
for i in range(15):
    if entries[i] % 2 == 0:
        even_numbers.append(entries[i])
        if len(even_numbers) == 5:
            for j in range(5):
                print(f"par[{j}] = {even_numbers[j]}")
            even_numbers = []
    else:
        odd_numbers.append(entries[i])
        if len(odd_numbers) == 5:
            for j in range(5):
                print(f"impar[{j}] = {odd_numbers[j]}")
            odd_numbers = []
            
# Printing any remaining odd numbers
for j in range(len(odd_numbers)):
    print(f"impar[{j}] = {odd_numbers[j]}")
# Printing any remaining even numbers
for j in range(len(even_numbers)):
    print(f"par[{j}] = {even_numbers[j]}")
    
# By: Fernando Serra
# https://github.com/fernandosserra

File: C/beginner/1179.c
// https://judge.beecrowd.com/pt/problems/view/1179
// Preenchimento de Vetor IV

// Libraries
#include <stdio.h>

// Main function
int main() {
    int entries[15], even[5], odd[5];
    int evenIndex, oddIndex, i, j;

    // Initialize indices for even and odd arrays
    evenIndex = 0;
    oddIndex = 0;
    i = 0;
    
    // Read 15 integers

    for(i = 0; i < 15; i++) {
        scanf("%d", &entries[i]);
    }

    // Distribute the integers into even and odd arrays
    for (i = 0; i < 15; i++) {
        if (entries[i] % 2 == 0) {  // Check if the number is even
            even[evenIndex++] = entries[i];
            if (evenIndex == 5) {
                for (j = 0; j < 5; j++) {
                    printf("par[%d] = %d\n", j, even[j]);
                }
                evenIndex = 0; // Reset index after printing
            }
        } else {
            odd[oddIndex++] = entries[i];
            if (oddIndex == 5) {
                for (j = 0; j < 5; j++) {
                    printf("impar[%d] = %d\n", j, odd[j]);
                }
                oddIndex = 0; // Reset index after printing
            }
        }
    }
    
    // Printing remaining odd numbers if there are any
    for (j = 0; j < oddIndex; j++) {
        printf("impar[%d] = %d\n", j, odd[j]);
    }
    
    // Printing remaining even numbers if there are any
    for (j = 0; j < evenIndex; j++) {
        printf("par[%d] = %d\n", j, even[j]);
    }    

    return 0;
}

// By: Fernando Serra
// https://github.com/fernandosserra